util/util.go

```diff
@@ -90,3 +93,18 @@ func RecoverIndexFromNodeName(name string) int {
 	}
 	return out
 }
+
+func GenerateFieldSelectorFromWorkflowName(wfName string) string {
+	result := fields.ParseSelectorOrDie(fmt.Sprintf("metadata.name=%s", wfName)).String()
+	compare := RecoverWorkflowNameFromSelectorString(result)
+	if wfName != compare {
+		panic(fmt.Errorf("Could not generate field selector from workflow name.\n"))
```

Comment:
```suggestion
		panic(fmt.Sprintf("Could not recover field selector from workflow name. Expected '%s' but got '%s'", wfName, compare))
```
Let's be more verbose with the panic. It could save us time in the future.

Reply:
Also, a neat trick you can do in Go is to call a method and use its result in the same line in an `if` statement.

```go
if compare := RecoverWorkflowNameFromSelectorString(result); compare != wfName {
   //...
}
```